rules/common.js

```diff
@@ -0,0 +1,42 @@
+module.exports = {
+  env: {
+    browser: true,
+    es6: true,
+    mocha: true,
+    jasmine: true,
+    node: true,
+    commonjs: true,
+  },
+  parserOptions: {
+    ecmaVersion: 2018,
+    sourceType: 'module',
+  },
+  rules: {
+    // -- Error
+    'no-shadow': 2,
+    'default-case': 2,
+    'max-params': ['error', 4],
+
+    // -- Warning
+    'class-methods-use-this': 1,
+    'no-else-return': 1,
+    'no-param-reassign': 1,
+    'no-underscore-dangle': 1,
+    'no-unused-vars': 1,
+    semi: [1, 'always'],
+    'import/no-unresolved': 1,
+    'import/prefer-default-export': 1,
+    'prefer-destructuring': 1,
+    'lines-between-class-members': 1,
+    'consistent-return': 1,
+    'max-lines': ['warn', { max: 500, skipBlankLines: true, skipComments: true }],
```

Comment:
This probably could be cut in half and even then that is a tad high. I believe the intention is to dial it back, but maybe we make a guesstimate around what the ideal file length should be and then adjust accordingly from there.